File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        for (String str : strs) {
            int[] arr = new int[26];
            for (char c : str.toCharArray()) {
                arr[c - 'a'] ++;
            }
            String s = Arrays.toString(arr);
            List<String> l = map.getOrDefault(s, new ArrayList<String>());
            l.add(str);
            map.put(s, l);
        }
        List<List<String>> res = new ArrayList<>();
        for (List<String> val : map.values()) {
            res.add(val);
        }
        return res;
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-0.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        res = ""
        for s in strs:
            length = len(s)
            res += f"#{length}{s}"
        return res
        
    def decode(self, s: str) -> List[str]:
        res = []
        for i in range(len(s)):
            c = s[i]
            if c == "#":
                length = int(s[i+1]) - int("0")
                res.append(s[(i+2):(i+2+length)])
                i += length + 1
        return res
                
